### DISCUSSION TITLE
Hide Already Installed Search Filters

### DISCUSSION THREAD

**Discussion by Vesperindustrial** [BlafKing/sd-civitai-browser-plus#107] [Ideas]
Unsure if this is at all possible but a way to have negative search filters on tags for example something like '-Characters' or something so you only get Styles, etc. Another would be dont show already installed items in the search.

**Comment by BlafKing** [comment]
Thanks for writing in!
This would have to be implemented from CivitAI's side so unfortunately I don't think this will be possible.

The [CivitAI REST API](https://github.com/civitai/civitai/wiki/REST-API-Reference) returns any results based on the provided parameters, and it's not even possible to provide multiple tags, let alone negative tags.
It's also unfortunately not possible to exclude installed models from the search results entirely, since the detection to check if models are installed is run locally.

It would be possible to simply hide any installed models but, that wouldn't make it alter the search results so personally I wouldn't find this to have much use :,)

**Reply by Vesperindustrial** [reply]
No worries, I really wasn't too sure how that all worked. The hide install models is more of a 'if i want to show 100 search results per page' then its visually less taxing to not actually see all the cards that page would return. My imagination was that it was more of a client side toggle, the actual search results wouldnt necissarily change. But I totally understand if you dont think thats a helpful scope option. Appreciate the reply!

**Reply by BlafKing** [reply]
Thanks for clarifying that you intended for it to be a client side toggle! That should definitely be possible to implement.

I personally don't care too much if I deem something to be useful or not, that's what the toggles are for :)
The more options the better, as long as you find it to be useful then that's good enough for me.

I'll make sure to add a toggle in the next update which you can use to hide any installed models! 👍🏻

**Comment by BlafKing** [comment]
I've now added a toggle in the filter dropdown to hide any installed models in v3.1.0! :)
